pkg/platform/kube/controller/nucliofunction.go

```diff
@@ -177,9 +179,12 @@ func (fo *functionOperator) CreateOrUpdate(ctx context.Context, object runtime.O
 		defer cancel()
 
 		// wait until the function resources are ready
-		if err = fo.functionresClient.WaitAvailable(waitContext, function.Namespace, function.Name); err != nil {
+		if err = fo.functionresClient.WaitAvailable(waitContext,
+			function.Namespace,
+			function.Name,
+			functionResourcesCreateOrUpdateTimestamp); err != nil {
 			return fo.setFunctionError(function,
-				functionconfig.FunctionStateUnhealthy,
+				functionconfig.FunctionStateError,
```

Comment:
@liranbg 
We can maybe check deployment failure **after** wait available failed, and if it is set function state to **Error**.
Otherwise, set function state to **Unhealthy**.
What do you think?